src/main/java/org/opensearch/security/dlic/rest/api/AbstractApiAction.java

```diff
@@ -113,150 +128,215 @@ protected AbstractApiAction(
+        );
     }
 
-    protected void handlePut(final RestChannel channel, final RestRequest request, final Client client, final JsonNode content)
-        throws IOException {
-        final String name = request.param("name");
-        if (name == null || name.length() == 0) {
-            badRequestResponse(channel, "No " + getResourceName() + " specified.");
-            return;
-        }
-        final SecurityDynamicConfiguration<?> existingConfiguration = load(getConfigName(), false);
-        if (existingConfiguration.getSeqNo() < 0) {
-            forbidden(
-                channel,
-                "Security index need to be updated to support '" + getConfigName().toLCString() + "'. Use SecurityAdmin to populate."
+    protected final ValidationResult<SecurityDynamicConfiguration<?>> loadConfiguration(
+        final CType cType,
+        final boolean logComplianceEvent
+    ) {
+        final var configuration = load(cType, logComplianceEvent);
+        if (configuration.getSeqNo() < 0) {
+            return ValidationResult.error(
+                RestStatus.FORBIDDEN,
+                forbiddenMessage(
+                    "Security index need to be updated to support '" + getConfigName().toLCString() + "'. Use SecurityAdmin to populate."
+                )
             );
-            return;
         }
+        return ValidationResult.success(configuration);
+    }
 
-        if (!isWriteable(channel, existingConfiguration, name)) {
-            return;
-        }
+    protected final ValidationResult<SecurityConfiguration> resourceExists(final SecurityConfiguration securityConfiguration) {
+        return resourceExists(getResourceName(), securityConfiguration);
+    }
 
-        if (isReadonlyFieldUpdated(existingConfiguration, content)) {
-            conflict(channel, "Attempted to update read-only property.");
-            return;
+    protected final ValidationResult<SecurityConfiguration> resourceExists(
+        final String resourceName,
+        final SecurityConfiguration securityConfiguration
+    ) {
+        if (securityConfiguration.maybeResourceName().isPresent()) {
+            if (!securityConfiguration.resourceExists()) {
+                return ValidationResult.error(
+                    RestStatus.NOT_FOUND,
+                    notFoundMessage(resourceName + " '" + securityConfiguration.resourceName() + "' not found.")
+                );
+            }
+            return ValidationResult.success(securityConfiguration);
         }
+        return ValidationResult.success(securityConfiguration);
+    }
 
-        if (LOGGER.isTraceEnabled() && content != null) {
-            LOGGER.trace(content.toString());
-        }
+    protected final ValidationResult<SecurityConfiguration> resourceCanBeChanged(final SecurityConfiguration securityConfiguration)
+        throws IOException {
+        return resourceNotHidden(securityConfiguration).map(ignore -> resourceNotReadOnly(securityConfiguration));
```

Comment:
Do static or reserved need to be considered here?

Reply:
Both, but I will change the function body. There is no need to use chaining here.

Reply:
Address in the latest commit. 
All configuration validation extracted into a separate class: `ConfigurationValidators` together with the test `ConfigurationValidatorsTest`.

Reply:
Changed it.